### PR TITLE
feat(Framework): update to ui5-webcomponents-rc.6

### LINKED ISSUE #349: Update UI5 Web Components to rc.6
Closes #343 
Closes #337 
Closes #305 
Closes #299 
Closes #364
Closes #306 
Closes #353

### PR DESCRIPTION
**Pull request by MarcusNotheis**
## Features
Please look at the [UI5 Web Components Changelog](https://github.com/SAP/ui5-webcomponents/blob/master/CHANGELOG.md#100-rc6-2020-03-27).

## Breaking Changes
BREAKING CHANGE: Replace Avatar Component with `Avatar` UI5 Web Component
BREAKING CHANGE: Replace Carousel Component with `Carousel` UI5 Web Component
BREAKING CHANGE: Replace SegmentedButton Component with `SegmentedButton` UI5 Web Component
BREAKING CHANGE: Deleted SegmentedButtonItem, replaced by ToggleButton
BREAKING CHANGE: Remove MessageToast Component, replaced by `Toast` UI5 Web Component
BREAKING CHANGE: **ThemeProvider**: Remove prop `withToastContainer` 
BREAKING CHANGE: **AvatarSize**: Remove value `Custom`
BREAKING CHANGE: **ButtonDesign**: Rename`Accept` to `Positive`
BREAKING CHANGE: **ButtonDesign**: Rename`Reject` to `Negative`
BREAKING CHANGE: **InputType**: Rename `Url` to `URL`
BREAKING CHANGE: **CarouselArrowsPlacement**: Rename `PageIndicator` to `Navigation`
BREAKING CHANGE:**BusyIndicator** : change default size of busy dots to `Medium`
BREAKING CHANGE:**Icon** : `src` property is removed in favor of `name`
BREAKING CHANGE:**MessageStrip** : replace icon property with icon slot to allow setting arbitrary content by the user
BREAKING CHANGE:**MultiComboBox** : Does not longer accept `StandardListItem` as child, use `MultiComboBoxItem` instead
BREAKING CHANGE:**ShellBar** : profile property is removed in favour of profile slot
BREAKING CHANGE:**TabContainer** : `onItemSelect` event is renamed to `onTabSelect` and the item event parameter is renamed to tab.
BREAKING CHANGE:**Card** : the property `subtitle` has been renamed to `subheading`
BREAKING CHANGE:**TextArea** : the property `maxLength` has been renamed to `maxlength`
BREAKING CHANGE:**ThemeProvider** : new prop `theme` is required for correct theming in addition to importing `import '@ui5/webcomponents-react/lib/ThemingSupport'`
BREAKING CHANGE: configuring of compact size is removed, use the `ui5-content-density-compact` CSS class to apply compact size.

Closes #349

### CODE REVIEW

File: packages/main/src/webComponents/Card/Card.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  component: Card
+};
+
+export const generatedDefaultStory = () => (
+  <Card
+    headerInteractive={boolean('headerInteractive', false)}
+    heading={text('heading', 'Heading')}
+    status={text('status', '1 of 5')}
+    subheading={text('subheading', 'Active Items')}
+    avatar={null}
```
Comment (Lukas742) [inline packages/main/src/webComponents/Card/Card.stories.tsx]: Add example:
```suggestion
    avatar={<Avatar backgroundColor="Accent3" icon="person-placeholder" />}
```

File: packages/main/src/webComponents/Card/Card.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { action } from '@storybook/addon-actions';
+import { boolean, text } from '@storybook/addon-knobs';
+import { Card } from '@ui5/webcomponents-react/lib/Card';
```
Comment (Lukas742) [inline packages/main/src/webComponents/Card/Card.stories.tsx]: Add imports for example:
```suggestion
import '@ui5/webcomponents-icons/dist/icons/person-placeholder.js';
import { Avatar } from '@ui5/webcomponents-react/lib/Avatar';
import { Card } from '@ui5/webcomponents-react/lib/Card';
```

File: packages/main/src/webComponents/CheckBox/CheckBox.stories.tsx [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
   title: 'UI5 Web Components / CheckBox',
   component: CheckBox
 };
 
-export const defaultStory = () => (
+export const generatedDefaultStory = () => (
   <CheckBox
+    checked={boolean('checked', false)}
     disabled={boolean('disabled', false)}
+    name={text('name', '')}
```
Comment (Lukas742) [inline packages/main/src/webComponents/CheckBox/CheckBox.stories.tsx]: Missing `@ui5/webcomponents/dist/features/InputElementsFormSupport.js` import

File: packages/main/src/webComponents/Input/Input.stories.tsx [hunk 4]
```diff
@@ -1,33 +1,39 @@
...
+    subcomponents: { SuggestionItem }
+  }
 };
 
 export const generatedDefaultStory = () => (
   <Input
     disabled={boolean('disabled', false)}
-    placeholder={'A placeholder text...'}
+    maxlength={number('maxlength', undefined)}
+    name={text('name', '')}
```
Comment (Lukas742) [inline packages/main/src/webComponents/Input/Input.stories.tsx]: Add import for `name` prop: `@ui5/webcomponents/dist/features/InputElementsFormSupport.js`

**Comment by Lukas742** [comment]
- [ ] Carousel doesn't show content for index > 0 if `dir="rtl"`

- [ ] SegmentedButton `dir="rtl"` breaks layout.

- [ ] Card `dir="rtl"` breaks layout.

- [ ] `name` prop needs `@ui5/webcomponents/dist/features/InputElementsFormSupport.js` import

- [ ] DatePicker and Calendar: Props `maxDate`, `minDate`, `primaryCalendarType`, `selectedDates` are throwing errors when set.

- [ ] MessageStrip: Icon stays on left position if `dir="rtl"` is set

File: packages/main/src/webComponents/Switch/Switch.stories.tsx [hunk 5]
```diff
@@ -12,9 +12,9 @@ export const generatedDefaultStory = () => (
   <Switch
     checked={boolean('checked', false)}
     disabled={boolean('disabled', false)}
-    textOn={text('textOn', 'Yes')}
-    textOff={text('textOff', 'No')}
-    graphical={boolean('graphical', false)}
+    graphical={text('graphical', false)}
```
Comment (Lukas742) [inline packages/main/src/webComponents/Switch/Switch.stories.tsx]: `graphical` expects to be a boolean.
 ```suggestion
     graphical={boolean('graphical', false)}
```

**Comment by MarcusNotheis** [comment]
> * [ ]  whats happening with the ComboBox story?
What exactly are you referring to @vbersch?

**Comment by vbersch** [comment]
> > * [ ]   whats happening with the ComboBox story?
> >   What exactly are you referring to @vbersch?

I was referring to the storybook shortcuts 't' and 's' that we talked about. Is there anything we can do about this at the moment? Thats a storybook bug, isn´t it?

File: packages/main/src/webComponents/ComboBox/ComboBox.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  component: ComboBox,
+  parameters: {
+    subcomponents: { ComboBoxItem }
+  }
+};
+
+export const generatedDefaultStory = () => (
+  <ComboBox
+    disabled={boolean('disabled', false)}
+    filter={text('filter', 'StartsWithPerTerm')}
```
[the remark this answers is not in the document]
Comment (MarcusNotheis) [reply packages/main/src/webComponents/ComboBox/ComboBox.stories.tsx]: not a enum in UI5 Web Components, therefore also no enum on our side.

File: packages/main/src/webComponents/Calendar/Calendar.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+export default {
+  title: 'UI5 Web Components / Calendar',
+  component: Calendar
+};
+
+export const generatedDefaultStory = () => (
+  <Calendar
+    maxDate={text('maxDate', '')}
+    minDate={text('minDate', '')}
+    primaryCalendarType={text('primaryCalendarType', undefined)}
```
[the remark this answers is not in the document]
Comment (MarcusNotheis) [reply packages/main/src/webComponents/Calendar/Calendar.stories.tsx]: Disabled this knob as it is not possible to change the locale during runtime

**Pull request merged**